### DISCUSSION TITLE
A Finally step in run commands

### DISCUSSION THREAD

**Discussion by stckcrsh** [nrwl/nx#6685] [Feature Requests]
We have been using the run commands as a way to kick off our VR testing framework in our buildsystem (buildkite right now).  Currently when VR testing fails we create a tarball out of all the generated diffs and publish it, this has to be done in custom scripts, but if we had something like a finally step in run commands that would run on success or failure we could do this all in one step.

Basically what i am proposing is to add to the run-commands executor a `finally` prop.
```json
{
  "builder": "@nrwl/workspace:run-commands",
  "options": {
    "commands": [
      "nx run something-that-fails:fail"
    ],
    "finally":{
      "exitStatus":{
        "*": [
          "tar some directory"
        ]
      },
      "passThroughFailure": true
    },
    "parallel": false
  }
}
```

finally would be an object with properties of
* exitStatus - which is an object with keys "*" | number and would run the commands in the array.  
* passThroughFailure - would be wether the final status would be based of the first command blocks or if you use the exit status of the finally commands

I would love to see this, but also would like to hear if this is even a good idea.

**Comment by MrChrisRodriguez** [comment]
Agreed. We have a bunch of set-up tasks that we want to make sure we close out regardless of whether or not the intermediary steps succeed or fail (like killing pm2 tasks, for example).

**Comment by gioragutt** [comment]
I think it may be not too difficult to create a custom executor that runs `run-commands` under the hood and adds your intended functionality
